feat: kick off the Python course (#1023)
apify/apify-docs: sources/academy/webscraping/scraping_basics_python/index.md

This course is incomplete. As we work on adding new lessons, we would love to hear your feedback. Comment right here under each page or [file a GitHub Issue](https://github.com/apify/apify-docs/issues) to discuss a problem.

:::

In this course we'll use Python to create an application for watching prices. It'll be able to scrape all product pages of an e-commerce website and record prices. Data from several runs of such program would be useful for seeing trends in price changes, detecting discounts, etc.

<!--
TODO image of warehouse with some CVS or JSON exported, similar to sources/academy/webscraping/scraping_basics_javascript/images/beginners-data-collection.png, which is for some reason the same as sources/academy/webscraping/scraping_basics_javascript/images/beginners-data-extraction.png
-->

![E-commerce listing on the left, JSON with data on the right](./images/scraping.png)

## What you'll do

- Inspect pages using browser DevTools
- Download web pages using the HTTPX library
- Extract data from web pages using the Beautiful Soup library
- Save extracted data in various formats, e.g. CSV which MS Excel or Google Sheets can open
- Follow links programatically (crawling)

> - Save time and effort with frameworks, such as Scrapy, and scraping platforms, such as Apify

Reviewer: Nitpick: About the lists of bullet points - I would use either a small letter at the beginning, with no punctuation at the end (or comma). Or a capital letter at the beginning and punctuation at the end.

Author: I have no idea how this is done correctly in English and whether there are any rules about it, so I just freestyle my way through it. I know in Czech there's pretty exact rule about capital letters and punctuation in lists like this, and in Czech text I tend to follow it. I'll search the internet and if there's such a rule, I'll fix all bullet point lists to be consistent with it. Maybe @TC-MO can also advise.

Author: I did [some reading](https://proofed.com/writing-tips/how-to-punctuate-and-capitalize-bullet-points/) and it seems the rules are quite loose. I opt for having full sentences with a capital letter and a punctuation at the end, and fragments of sentence to start with lower case and end with something like comma (which is the Czech rule I'm basically following elsewhere in the course).

Author: Addressed in 531ca173

Reviewer 2: oh somehow this ping eluded me... either way, both are fine though one thing I'll mention whenever such lists are composed to try and remember to follow Parallel Structure as well.

Author: First time I hear about Parallel Structure, I'll check it out, thanks!

> - Save time and effort with frameworks, such as Scrapy, and scraping platforms, such as Apify

Author: I wrote this sentence before Crawlee for Python has existed 😀 We talked about it with @souravjain540 at EuroPython and I tend to agree we should choose Crawlee, even though it's very new, because it's just easier to grasp.

Ideally, Crawlee for Python docs would feature pages like "Crawlee vs Scrapy" or "Migrating from Scrapy", which would explain how to write the same thing in both, or guide people like me who are fluent in Scrapy and want to try or migrate to Crawlee. First idea was it could be a separate course in the academy, but after giving it a thought I'd say its best home is the Crawlee docs.

> - Save time and effort with frameworks, such as Scrapy, and scraping platforms, such as Apify

Author: I wouldn't use both in the course for beginners. First I think the code could be so different than even using tabs could result in a mess, second I strongly believe beginners diving to a new topic need to be shown a path without crossroads and options, and the author of the course is responsible for making all the hard choices for them.

> - Save time and effort with frameworks, such as Scrapy, and scraping platforms, such as Apify

Author: Addressed in https://github.com/apify/apify-docs/pull/1023/commits/307121a1a501105a6a323e63ffdc559caa42a628

## Who this course is for

Anyone with basic knowledge of developing programs in Python who wants to start with web scraping can take this course. The course does not expect you to have any prior knowledge of web technologies or scraping.

> ## Requirements
> 
> - macOS, Linux or Windows machine with a web browser and Python installed
> - Familiar with Python basics: variables, conditions, loops, functions, strings, lists, dictionaries, files, classes, exceptions
> - Comfortable importing from the Python standard library, using virtual environments, and installing dependencies with `pip`
> - Familiar with how to run commands in Terminal or Command Prompt

Reviewer 2: ```suggestion
## Requirements

- A macOS, Linux, or Windows machine with a web browser and Python installed
- Familiarity with Python basics: variables, conditions, loops, functions, strings, lists, dictionaries, files, classes, and exceptions
- Comfort with importing from the Python standard library, using virtual environments, and installing dependencies with `pip`
- Familiarity with running commands in Terminal (macOS/Linux) or Command Prompt (Windows)
```

Slight rewrite for parallel structure in list.

Author: Thanks!
